openverse_catalog/dags/providers/provider_reingestion_workflows.py

```diff
@@ -46,24 +46,19 @@ class ProviderReingestionWorkflow(ProviderWorkflow):
 
     def __post_init__(self):
         if not self.dag_id:
-            self.dag_id = f"{self.provider_script}_reingestion_workflow"
+            # Call super() first to initialize the provider_name
+            super().__post_init__()
+            # Override the dag_id
+            self.dag_id = f"{self.provider_name}_reingestion_workflow"
+            return
+
+        super().__post_init__()
```

Comment:
Do we need to call `super().__post_init__()` twice? 😮

Reply:
Yep, unfortunately. If a `dag_id` wasn't provided in the configuration, we want to override it. We need to call super first in order to set the `provider_name` (which is used in building the dag_id), and _then_ we can override it. We can't do this check _after_ calling super, because `dag_id` will have been set (to the regular `_workflow` name) during the super call. So we must call super within the branch.

But if `dag_id` was provided, we still want to call super(), so we also have to do it outside the branch. The return on L53 prevents it from actually being run twice.

Reply:
Oh, oops! Okay I see, we're not actually calling it twice as you point out. Thanks!